### ISSUE TITLE
On What to Watch page (maybe other pages), YouTubeCenter for Chrome consumes excessive CPU.

### ISSUE THREAD

**Issue by meldroc** [YePpHa/YouTubeCenter#1955]
I am currently running YouTube Center Developer Build 529, installed using TamperMonkey, on Chrome 43.0.2357.134, running on Kubuntu Linux 15.04.

I habitually keep a Youtube tab open and pinned, usually on the What To Watch page, or the home page for my Youtube account. Usually, what happens is that Youtube Center seems to work correctly at first, and I can surf, click around, watch videos.

However, if I just let the page sit, say if I go browse another tab, and then come back some time later, I notice that my machine's fans are running fast, and I check using a CPU meter app, and find that one of my CPU cores is at 100% usage. When I go into Chrome, I just hit Shift-Esc to summon Chrome's task manager, and I find my Youtube page is inexplicably eating one of my CPUs. Closing the tab and re-opening Youtube fixes the problem, temporarily...

I get the impression that somewhere, Youtube Center is caught in an infinite loop. Maybe it's spinning trying to find a resource that was available, but is no longer, say because the contents of What To Watch got changed (which Google does all the time.)

How to reproduce this error? Surf to your Youtube home page, and go to What To Watch, then go surf on another tab, and come back after a while. Try using the page, especially scrolling down, which induces it to load more videos. I've found that a combination of letting the page sit for a long time, then trying to scroll down, or use Youtube Center features, say by mousing over videos to show their resolutions, & votes, will often trigger this bug.

**Comment by caeldom** [comment]
I have a similar problem, upon loading the What to Watch page, the page will load fine for a few seconds, then freeze for another few seconds, before it is fully loaded. This only occurs on the What to Watch page. I haven't left it on long enough to notice any lag or cpu consumption after it's fully loaded though.

**Comment by meldroc** [comment]
caeldom - I'm noticing that too. The What To Watch page loads, then the page freezes for a while, presumably while Youtube Center is processing stuff, then eventually unfreezes

**Comment by SuperSajuuk** [comment]
Yes, there are several pages that just freeze up with YT Center now. You aren't alone with this and the persistent freezing, with nothing being done by the developer to sort it, forced me to stop using the script. The only time the script didn't freeze on page loads, was the short time after an update had been performed, which didn't last very long. I have no idea if it's a Tampermonkey problem, or it's actually a problem with the script itself.

I don't see the issue of excessive CPU / RAM usage being solved any time soon.

**Comment by caeldom** [comment]
Good to know others are experiencing the same issue!
